apps/web/pages/settings/my-account/profile.tsx

```diff
@@ -317,6 +300,16 @@
           resendVerifyEmailMutation.mutate({ email });
           showToast(t("email_sent"), "success");
         }}
+        handleAccountDisconnect={(values) => {
+          if (!isCALIdentityProvider) {
+            if (!user?.passwordAdded) {
+              setShowCreateAccountPasswordDialog(true);
+            } else {
+              setTempFormValues(values);
+              setShowAccountDisconnectWarning(true);
+            }
+          }
+        }}
```

Comment:
[Prefer Early Returns](https://handbook.cal.com/engineering/best-practices/prefer-early-returns) 

```suggestion
        handleAccountDisconnect={(values) => {
          if (isCALIdentityProvider) return;
          if (user?.passwordAdded) {
            setTempFormValues(values);
            setShowAccountDisconnectWarning(true);
            return;
          }
          setShowCreateAccountPasswordDialog(true);
        }}
```